Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/ConnectionManagerBean.java

```diff
@@ -0,0 +1,273 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.admingui.managed;
+
+import gov.hhs.fha.nhinc.admingui.application.EndpointManagerCache;
+import gov.hhs.fha.nhinc.admingui.model.ConnectionEndpoint;
+import gov.hhs.fha.nhinc.admingui.services.PingService;
+import gov.hhs.fha.nhinc.admingui.services.impl.PingServiceImpl;
+import gov.hhs.fha.nhinc.connectmgr.ConnectionManagerCache;
+import gov.hhs.fha.nhinc.connectmgr.ConnectionManagerException;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import gov.hhs.fha.nhinc.properties.PropertyAccessException;
+import gov.hhs.fha.nhinc.properties.PropertyAccessor;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.ViewScoped;
+import org.apache.http.impl.cookie.DateUtils;
+import org.apache.log4j.Logger;
+import org.uddi.api_v3.BindingTemplate;
+import org.uddi.api_v3.BusinessEntity;
+import org.uddi.api_v3.BusinessService;
+import org.uddi.api_v3.CategoryBag;
+import org.uddi.api_v3.Contact;
+import org.uddi.api_v3.KeyedReference;
+
+/**
+ *
+ * @author jassmit
+ */
+@ManagedBean(name = "connectionManagerBean")
+@ViewScoped
+public class ConnectionManagerBean {
+
+    private static final Logger LOG = Logger.getLogger(ConnectionManagerBean.class);
+    
+    private final HashMap<String, BusinessEntity> externalEntities = new HashMap<String, BusinessEntity>();
+    private final List<String> entityNames = new ArrayList<String>();
+    
+    //TODO Display local entity endpoints (internal and external).
+    private BusinessEntity localEntity;
+    
+    private BusinessEntity selectedEntity;
+    private String selectedEntityName;
+    
+    private List<ConnectionEndpoint> endpoints;
+    private ConnectionEndpoint selectedEndpoint;
+    
+    private final PingService pingService = new PingServiceImpl();
+    
+    private static final String HCID_TMODEL_KEY = "uddi:nhin:nhie:homecommunityid";
+    private static final String REGION_TMODEL_KEY = "uddi:uddi.org:ubr:categorization:iso3166";
+    private static final String DATE_FORMAT = "MM-dd-yy hh:mm:ss";
+    
+    @PostConstruct
+    public void init() {
+        refresh();
+    }
+    
+    public List<String> getEntityNames(){
+        return entityNames;
+    }
+    
+    public String getSelectedEntityName() {
+        String name = "--";
```

Comment:
All standalone instances of "--" should be pulled out into a constant (e.g., "NULL_DISPLAY") so that we have the flexibility to change the display value later.